# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../psmisc/no-glibcism.patch
# Copyright (C) 2006 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

Though I have not looked it up in standards this looks quite like a glibc'ism,
of course dietlibc does not include this crap and other C libraries might
neither.

  - Rene Rebe <rene@exactcode.de>

--- psmisc-21.9/src/pstree.c.vanilla	2006-01-06 23:19:57.000000000 +0100
+++ psmisc-21.9/src/pstree.c	2006-01-06 23:22:53.000000000 +0100
@@ -48,8 +48,6 @@
 #include <selinux/selinux.h>
 #endif /*WITH_SELINUX*/
 
-extern const char *__progname;
-
 #ifndef MAX_DEPTH
 #define MAX_DEPTH    100
 #endif
@@ -810,7 +808,7 @@
   textdomain(PACKAGE);
 #endif
   
-  if (!strcmp(__progname, "pstree.x11"))
+  if (strstr(argv[0], "pstree.x11"))
     wait_end=1;
 
   /*
